xchange-bitmex/src/main/java/info/bitrich/xchangestream/bitmex/BitmexStreamingMarketDataService.java

```diff
@@ -1,53 +1,70 @@
 package info.bitrich.xchangestream.bitmex;
 
-import info.bitrich.xchangestream.bitmex.dto.BitmexLimitOrder;
-import info.bitrich.xchangestream.bitmex.dto.BitmexOrderbook;
-import info.bitrich.xchangestream.bitmex.dto.BitmexTicker;
-import info.bitrich.xchangestream.bitmex.dto.BitmexTrade;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.bitmex.dto.*;
 import info.bitrich.xchangestream.core.StreamingMarketDataService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
 import io.reactivex.Observable;
+import org.knowm.xchange.bitmex.BitmexExchange;
+import org.knowm.xchange.bitmex.BitmexPrompt;
 import org.knowm.xchange.currency.CurrencyPair;
 import org.knowm.xchange.dto.marketdata.OrderBook;
 import org.knowm.xchange.dto.marketdata.Ticker;
 import org.knowm.xchange.dto.marketdata.Trade;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.SortedMap;
-import java.util.TreeMap;
+import java.io.IOException;
+import java.util.*;
 
 /**
  * Created by Lukas Zaoralek on 13.11.17.
  */
 public class BitmexStreamingMarketDataService implements StreamingMarketDataService {
     private static final Logger LOG = LoggerFactory.getLogger(BitmexStreamingMarketDataService.class);
 
+    private final ObjectMapper objectMapper = StreamingObjectMapperHelper.getObjectMapper();
+
     private final BitmexStreamingService streamingService;
 
-    private final SortedMap<CurrencyPair, BitmexOrderbook> orderbooks = new TreeMap<>();
+    private final BitmexExchange bitmexExchange;
+
+    private final SortedMap<String, BitmexOrderbook> orderbooks = new TreeMap<>();
 
-    public BitmexStreamingMarketDataService(BitmexStreamingService streamingService) {
+    public BitmexStreamingMarketDataService(BitmexStreamingService streamingService, BitmexExchange bitmexExchange) {
         this.streamingService = streamingService;
+        this.streamingService.subscribeConnectionSuccess().subscribe(o -> {
+            LOG.info("Bitmex connection succeeded. Clearing orderbooks.");
+            orderbooks.clear();
+        });
+        this.bitmexExchange = bitmexExchange;
+    }
+
+    private String getBitmexSymbol(CurrencyPair currencyPair, Object... args) {
```

Comment:
This behaviour is now inconsistent with XChange following a change to XChange's API recently.  `currencyPair` should be simply `XBTUSD` or `XBTM19`. The `determineActiveContract` method is intended to be called before methods like this where either compatibility with the previous behaviour is required or you want your code to work across the expiry date of a contract.

Can we make this:

```
private String getBitmexSymbol(CurrencyPair currencyPair) {
  return currencyPair.base.toString() + currencyPair.counter.toString();
}
```